monitoring/system_pods.go

```diff
@@ -0,0 +1,248 @@
+				WithField("phase", pod.Status.Phase).
+				Debug("Pod is in an unknown phase.")
+		}
+	}
+}
+
+// isInitialized returns true if the pod's `Initialized` condition is `True`.
+func isInitialized(conditions []corev1.PodCondition) bool {
+	for _, condition := range conditions {
+		if condition.Type == corev1.PodInitialized && condition.Status == corev1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+// verifyContainers verifies all containers are in a healthy state.
+// Returns an error with state and reason if a container is in an unhealthy state.
+func verifyContainers(containerStatuses []corev1.ContainerStatus) error {
+	for _, status := range containerStatuses {
+		if status.State.Waiting != nil {
+			reason := status.State.Waiting.Reason
+			if reason == errImagePullBackOff || reason == errCrashLoopBackOff || reason == errImagePull {
+				return trace.BadParameter("%v waiting: %v", status.Name, reason)
+			}
+		}
+		if status.State.Terminated != nil {
+			if status.State.Terminated.ExitCode != 0 {
+				return trace.BadParameter("%v terminated: %v", status.Name, status.State.Terminated.Reason)
+			}
+		}
+	}
+	return nil
+}
+
+// systemPodsFailureProbe constructs a probe that represents a failed system pods
+// check for the pod specified by podName and namespace.
+func systemPodsFailureProbe(checkerName, namespace, podName string, err error) *pb.Probe {
+	return &pb.Probe{
+		Checker:  checkerName,
+		Detail:   fmt.Sprintf("pod %v/%v is not running", namespace, podName),
+		Error:    trace.UserMessage(err),
+		Status:   pb.Probe_Failed,
+		Severity: pb.Probe_Warning, // TODO: set probe to critical
+	}
+}
+
+const systemPodsCheckerID = "system-pods-checker"
+const systemPodKey = "gravitational.io/critical-pod"
+
+// NOTE: Currently, there is limited documentation on k8s container state.
+// Information on container state can be found in kubelet pkg:
+// https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet
+
+// These states usually do not indicate an unhealthy pod.
+const (
+	// podInitializing state indicates that the container is waiting on an initContainer to terminate.
+	podInitializing = "PodInitializing"
+	// containerCreating state indicates that the container is being created.
+	containerCreating = "ContainerCreating"
+	// containerCompleted state indicates that the container has terminated without error.
+	containerCompleted = "Completed"
+)
+
+// These states are indicative of a unhealthy pod.
+const (
+	// errCrashLoopBackOff state indicates that the container is in a crash loop.
+	errCrashLoopBackOff = "CrashLoopBackOff"
+	// errImagePullBackOff state indicates that the container image pull failed.
+	errImagePullBackOff = "ImagePullBackOff"
+	// errImagePull state indicates that the container image pull failed. General image pull error.
+	errImagePull = "ErrImagePull"
+)
+
+// containerError state indicates that the container terminated with an error.
+const containerError = "Error"
```

Comment:
This seems to be only used in tests?

Reply:
Yeah, I guess we don't need to keep this just for the tests.